packages/relay-compiler/index.d.ts

```diff
@@ -0,0 +1,332 @@
+/**
+ * This is the object containing the configuration of your Relay project.
+ * 
+ * @see https://github.com/facebook/relay/tree/main/packages/relay-compiler
+ */
+export type RelayConfig = RelayConfigSingleProject | RelayConfigMultiProject;
+
+/**
+ * This is the configuration containing the object for single project configuration.
+ */
+export type RelayConfigSingleProject = {
+  /** Root directory of application code. */
+  src: string;
+  /** Relative path to the file with GraphQL SDL file. */
+  schema: string;
+  /** The name of the language used for input files and generated artifacts. */
+  language: RelayConfigLanguage;
+  /** A specific directory to output all artifacts to. When enabling this the babel plugin needs `artifactDirectory` to be set as well. */
+  artifactDirectory?: string;
+  /**
+   * Directories to ignore under `src`.
+   * 
+   * @default ["/node_modules/", "/mocks/", "/generated/"]
+   */
+  excludes?: string[];
+  /** Extra config for the schema itself. */
+  schemaConfig?: RelayConfigSchemaConfig;
+  /**
+   * This option controls whether or not a catch-all entry is added to enum type definitions values that may be added 
+   * in the future. Enabling this means you will have to update your application whenever the GraphQL server schema adds 
+   * new enum values to prevent it from breaking.
+   * 
+   * @default false
+   */
+  noFutureProofEnums?: boolean;
+  /** Mappings from custom scalars in your schema to built-in GraphQL types, for type emission purposes. */
+  customScalars?: RelayConfigCustomScalars;
+  /**
+   * This option enables emitting ES modules artifacts.
+   * 
+   * @default false
+   */
+  eagerEsModules?: boolean;
+  /** Relay supports two versions of the config. */
+  persistConfig?: RelayConfigPersist;
+  /** Name of the command that runs the relay compiler */
+  codegenCommand?: string;
+  /**
+   * Name of the global variable for dev mode
+   * 
+   * @default "__DEV__"
+   */
+  isDevVariableName?: string;
+  /** 
+   * Formatting style for generated files.
+   * 
+   * @default "commonjs"
+   */
+  jsModuleFormat?: 'commonjs' | 'haste';
+  /** Options for configuring the output of compiler diagnostics. */
+  diagnosticReportConfig: RelayConfigDiagnosticReport;
```

Comment:
I think we need to understand defaults a little better, since I know a config without this value (for example, but others as well) is accepted by the compiler.

Ah, looking a bit more closely, it looks like the struct itself is annotated with `default`. Can you check each of the structs and see how that impacts these types?

https://github.com/facebook/relay/blob/ece1fd953b987d08ab1833d5c1f63dee7728a4ab/compiler/crates/relay-compiler/src/config.rs#L666

Reply:
I checked all the structs and update some JSDocs with new `@default` values. I think that everything is OK now, what do you think?